scripts/engineering/generate_phase44d_carry_aware_experimental_2x2_scaling_figure.py

```
@@ -0,0 +1,308 @@
+        raise SystemExit(f"no rows found in {path}")
+    return rows
+
+
+def timing_metadata(rows: list[dict[str, str]], *, fallback_runs: int) -> tuple[str, int]:
+    first = rows[0]
+    mode = first.get("timing_mode", "").strip()
+    policy = first.get("timing_policy", "").strip()
+    unit = first.get("timing_unit", "").strip()
+    runs_raw = first.get("timing_runs", "").strip()
+    if not mode:
+        raise SystemExit("phase44d engineering benchmark rows must include timing_mode")
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)